### PR TITLE
CAB-4075: Adds E2E test for Bulk Update feature.

### PR DESCRIPTION
**Pull request by farmas**
There is only one E2E test here that verifies the whole user flow, other functionality is covered by unit tests.

Part of this PR is the beginnings of re-usable page objects: a page object for the SearchResultsComponent is created and used by several pages.

### CODE REVIEW

File: protractor.conf.local.js [hunk 1]
```diff
@@ -25,6 +25,6 @@ exports.config = {
   },
   onPrepare() {
     require('ts-node').register({project: 'e2e/tsconfig.e2e.json'});
-    jasmine.getEnv().addReporter(new SpecReporter({spec: {displayStacktrace: 'pretty'}}));
+    jasmine.getEnv().addReporter(new SpecReporter({spec: {displayStacktrace: 'raw'}}));
```
Comment (farmas) [inline protractor.conf.local.js:28]: I found the 'pretty' stack trace to be overwhelming and harder to parse, changed this to 'raw' for a more compact view.

File: e2e/search/search-results.po.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      return Array.from(new Set(results));
+    });
+  }
+
+  /**
+   * Sorts results by the column header with the text specified.
+   * @param headerText The text of the column to sort the results by.
+   */
+  sortByHeaderText(headerText: string): promise.Promise<any> {
+    const publishStatusHeader = this.root.element(by.buttonText(headerText));
```
Comment (huiaic) [inline e2e/search/search-results.po.ts:52]: Maybe rename `publishStatusHeader` to `columnHeader` to be generic?

File: e2e/search/search-results.po.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  /**
+   * Gets the rows of the results table.
+   */
+  get rows(): ElementArrayFinder {
+    return this.root.all(by.css('app-search-results .mat-row'));
+  }
+
+  /**
+   * Gets the vales of each row cell of a given column.
```
Comment (huiaic) [inline e2e/search/search-results.po.ts:27]:
```suggestion
   * Gets the values of each row cell of a given column.
```

File: e2e/search/search-results.po.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  sortByHeaderText(headerText: string): promise.Promise<any> {
+    const publishStatusHeader = this.root.element(by.buttonText(headerText));
+    return publishStatusHeader.click();
+  }
+
+  /**
+   * Selects rows specified by index.
+   * @param indeces Array of row indeces to select.
+   */
+  selectRows(...indeces: number[]): promise.Promise<any> {
```
Comment (huiaic) [inline e2e/search/search-results.po.ts:60]: My spell checker sense kicked in - should be "indices"

File: e2e/browserUtils.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    });
+  }
+
+  /**
+   * Waits for an element to contain the given text.
+   * @param elementOrSelector Element or selector to locate the element to test.
+   * @param expectedText Expected text.
+   * @param timeoutMilliseconds Timeout in milliseconds to wait for.
+   */
+  waitForElementText(elementOrSelector: string | ElementFinder, expectedText: string, timeoutMilliseconds: number = 5000): promise.Promise<any>  {
```
[the remark this answers is not in the document]
Comment (farmas) [reply e2e/browserUtils.ts:51]: The problem with the way protractor designed wait-for-conditions is that it does not provide current value in the error message (and as far as I know, no way to supply the current value in the message). As in, if the condition fails, the log will just say "condition not met expected 'bar'", but for maintainability I would like the log to say "condition not met, the last value was 'foo' expected 'bar'", which is what this type of functions try to address.

File: e2e/search/search-results.po.ts (hunk 4 above)
Comment (farmas) [reply e2e/search/search-results.po.ts:60]: Oh english, why you so hard!

File: e2e/browserUtils.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (huiaic) [reply e2e/browserUtils.ts:51]: I see, that's too bad re Protractor limitations. Thanks for the explanation.

**Pull request merged**